### ISSUE TITLE
Display Region on 'Search for Player' Textbox

### ISSUE THREAD

**Issue by BrandonCookeDev** [ripgarpr/garpr#106]
Pending Tournament screens are great, however one thing that hinders Importers is when more than one player from different regions has the same tag. In particular, the tag Blank came up more than 7 times and I had no idea which one was from my region.

We should try to display a player's region with their tag when the suggestion popup appears:

![whichblank](https://cloud.githubusercontent.com/assets/10892126/17951978/3a2bc2f6-6a34-11e6-9a89-0571c94be9d9.jpg)

**Comment by jschnei** [comment]
One easy solution is just for an admin to rename the player to include the region (e.g. "Blank (norcal)") if there are two players with the same tag from different regions (I think we do this right now for some players). I think this is mostly happening because admins are renaming players that don't exist (e.g. players from byes) to "Blank" since we don't have a delete player functionality yet (something we should probably make an issue for if there isn't one already).

**Comment by BrandonCookeDev** [comment]
True, my only concern is players being picky and complaining about the region in their name.

I figure this actually should be easily resolve since the PlayerService could potential query by {name, region}. I believe that's how it currently works if I'm not mistaken

**Comment by BrandonCookeDev** [comment]
@jschnei 
Assuming we pull back the entire player Model, is there any way to make this line:

![dropdownloop](https://cloud.githubusercontent.com/assets/10892126/18066009/6a4dfa2a-6e04-11e6-9cd3-18c75a1cb93d.PNG)

to read p.name - (p.region)

**Comment by jschnei** [comment]
I am not an expert on frontend stuff, but you could make some function "searchboxFormatPlayer" in app.js that returns "p.name - (p.region)", and then replace that with "p as searchboxFormatPlayer(p) for p in getMergePlayers($viewValue)".

**Comment by BrandonCookeDev** [comment]
So I actually got this yo work except for two things 
- players from None or Invalid regions don't seem to appear. 
- type head includes region in the alphabetical search 

Looking into fixes for these

**Comment by BrandonCookeDev** [comment]
actually I was wrong, Typeahead does NOT seem to include regions in the search.

@jessemtso @jschnei does our current dev db include anyone in regions None or Invalid?

**Comment by jschnei** [comment]
the current dev db definitely includes people in regions that are invalid; soon i'll update it to a dump of the current db soon that shouldn't have this issue.

**Comment by Jamesargy6** [comment]
@jschnei I've notice that players in the prod Database also sometimes have "invalid region" attributed to them, sometimes in addition to other "real" regions. Is this a seperate/known issue or no?

**Comment by BrandonCookeDev** [comment]
@Jamesargy6  
This is the result of old regions that have since been deleted. Most of these were probably SoCal

**Comment by BrandonCookeDev** [comment]
Oopsie

**Comment by BrandonCookeDev** [comment]
So I was having luck for this, then we ran into an issue of the textbox not filling in on enter or behaving correctly on the UI. The angular controller operates fine however, merging players should work and players are searched for.

However, i'm not comfortable submitting incomplete functionality for the PR.
Took a different route of inserting the TypeaheadLabel function (makes the text read Player Name ~ Region) in as a function of the PlayerService, then adding a typeaheadName property to each player object that is pushed to the filteredPlayers array that appears in the typeahead when you search. See below:

```
getPlayerListFromQuery: function(query, filter_fn) {
            var TYPEAHEAD_PLAYER_LIMIT = 20;
            var filteredPlayers = [];
            for (var i = 0; i < this.allPlayerList.players.length; i++) {
                var curPlayer = this.allPlayerList.players[i];

                curPlayer.typeaheadName = this.typeaheadLabel(curPlayer);
                if(filter_fn == null || filter_fn(curPlayer)){
                    var matchQuality = this.playerMatchesQuery(curPlayer, query);
                    if(matchQuality > 0){
                        filteredPlayers.push({'player': curPlayer,
                                              'quality': matchQuality});
                    }
                }
            }
```

```
typeaheadLabel: function(player){
            var region = '';
            try{
                region  = player.regions[0];
            } catch(err){
                region = 'None'
            }
            region = region.italics();
            region = region.fontcolor('#cccccc');
            return player.name + ' ~ ' + region;
        },
```

This, however, currently makes nothing appear in the typeahead. 

I think this enhancement is close. Really close.
I think we could use a more experienced frontend angular dev to assist with this and we can knock it out really quick.

**Comment by BrandonCookeDev** [comment]
I'm beginning to suspect that sending this as an extra property of the return object from the Python to the frontend will help avoid nearly all the current issues the JS is having

**Comment by BrandonCookeDev** [comment]
Should be fixed by #150 
Quick function in Players Service tacks on the typeahead value.
